From 0f29541c6728c9c2159080267b43e9b4144e9e56 Mon Sep 17 00:00:00 2001
From: Ludovic Rousseau <ludovic.rousseau@gmail.com>
Date: Tue, 8 May 2012 15:53:56 +0200
Subject: [PATCH 11/28] Core: Return LIBUSB_ERROR_IO on sync API
 TRANSFER_ERROR and _CANCELLED

This is much more appropriate and also fixes gcc -Wswitch-enum warnings.

libusbx.git commit 30ccbd773bc02d5dfd21e02bfba6d58234f59393
---
 libusb/sync.c |    8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/libusb/sync.c b/libusb/sync.c
index 8eed47b..ac3ab7e 100644
--- a/libusb/sync.c
+++ b/libusb/sync.c
@@ -135,6 +135,10 @@ int API_EXPORTED libusb_control_transfer(libusb_device_handle *dev_handle,
 	case LIBUSB_TRANSFER_OVERFLOW:
 		r = LIBUSB_ERROR_OVERFLOW;
 		break;
+	case LIBUSB_TRANSFER_ERROR:
+	case LIBUSB_TRANSFER_CANCELLED:
+		r = LIBUSB_ERROR_IO;
+		break;
 	default:
 		usbi_warn(HANDLE_CTX(dev_handle),
 			"unrecognised status code %d", transfer->status);
@@ -205,6 +209,10 @@ static int do_sync_bulk_transfer(struct libusb_device_handle *dev_handle,
 	case LIBUSB_TRANSFER_NO_DEVICE:
 		r = LIBUSB_ERROR_NO_DEVICE;
 		break;
+	case LIBUSB_TRANSFER_ERROR:
+	case LIBUSB_TRANSFER_CANCELLED:
+		r = LIBUSB_ERROR_IO;
+		break;
 	default:
 		usbi_warn(HANDLE_CTX(dev_handle),
 			"unrecognised status code %d", transfer->status);
-- 
1.7.9.5

